### ISSUE TITLE
Operate item orders don't clear or retarget after switching the subs

### ISSUE THREAD

**Issue by itchyOwl** [Regalis11/Barotrauma#11437]
The issue issue is that the order indicators show the orders as they were valid, but I think they still target the items on the previous sub. I think either a) the operate item orders should automatically clear when the item is unreachable, or b) we should programmatically clear all the orders after switching the subs.

![image.png](https://images.zenhubusercontent.com/5c93b12055cf186934fc0a03/350e00a2-a498-478e-a2b2-72903e1f6b80)

Currently the work-around to the issue is to clear and reissue the orders.

**Comment by Regalis11** [comment]
>  I think either a) the operate item orders should automatically clear when the item is unreachable, or b) we should programmatically

Looks like this was already implemented, and there was also some logic for finding a new target for the order when that's possible. It no longer worked though, because the submarine switch was handled before saving the orders, and PendingSubmarineSwitch was already null when we tried to determine if we've opted to switched the sub. Possibly broken by the changes done to the item transfer logic?

Fixed in https://github.com/Regalis11/Barotrauma-development/commit/2ae99600580fe84f87d8ab987732f2b999ef476b
